drivers/soundwire/cadence_master.c

```diff
@@ -1257,6 +1282,8 @@ int sdw_cdns_init(struct sdw_cdns *cdns)
 
 	cdns_init_clock_ctrl(cdns);
 
+	sdw_cdns_check_self_clearing_bits(cdns, __func__);
```

Comment:
just out of curiosity, Pierre, why is this function void? Why don't we return an error and abort init?

Reply:
@ranj063 For now I only wanted to run stress tests and see if we actually see this sort of problems. @bardliao mentioned that he saw this happening, and I just want to add this check to our development branch so that we get reports.

Changing the error flow, and/or optimizing the sleep time, is a possibility, but that's too much for what remains speculative bug chasing. We probably don't want to upstream this either until we see a real case where we've hit an issue.